File: go/extensions/signinwithx/schema.go
package signinwithx

// Schema returns the JSON Schema for validating SIWX client payloads.
func Schema() map[string]interface{} {
	return map[string]interface{}{
		"$schema": "https://json-schema.org/draft/2020-12/schema",
		"type":    "object",
		"properties": map[string]interface{}{
			"domain": map[string]interface{}{
				"type": "string",
			},
			"address": map[string]interface{}{
				"type": "string",
			},
			"statement": map[string]interface{}{
				"type": "string",
			},
			"uri": map[string]interface{}{
				"type":   "string",
				"format": "uri",
			},
			"version": map[string]interface{}{
				"type": "string",
			},
			"chainId": map[string]interface{}{
				"type": "string",
			},
			"type": map[string]interface{}{
				"type": "string",
			},
			"nonce": map[string]interface{}{
				"type": "string",
			},
			"issuedAt": map[string]interface{}{
				"type":   "string",
				"format": "date-time",
			},
			"expirationTime": map[string]interface{}{
				"type":   "string",
				"format": "date-time",
			},
			"notBefore": map[string]interface{}{
				"type":   "string",
				"format": "date-time",
			},
			"requestId": map[string]interface{}{
				"type": "string",
			},
			"resources": map[string]interface{}{
				"type": "array",
				"items": map[string]interface{}{
					"type":   "string",
					"format": "uri",
				},
			},
			"signature": map[string]interface{}{
				"type": "string",
			},
		},
		"required": []string{
			"domain",
			"address",
			"uri",
			"version",
			"chainId",
			"type",
			"nonce",
			"issuedAt",
			"signature",
		},
	}
}

File: typescript/packages/extensions/src/sign-in-with-x/schema.ts
/**
 * JSON Schema builder for SIWX extension
 *
 * Per CHANGELOG-v2.md lines 276-292
 */

import type { SIWxExtensionSchema } from "./types";

/**
 * Build JSON Schema for SIWX extension validation.
 * This schema validates the client proof payload structure.
 *
 * @returns JSON Schema for validating SIWX client payloads
 */
export function buildSIWxSchema(): SIWxExtensionSchema {
  return {
    $schema: "https://json-schema.org/draft/2020-12/schema",
    type: "object",
    properties: {
      domain: { type: "string" },
      address: { type: "string" },
      statement: { type: "string" },
      uri: { type: "string", format: "uri" },
      version: { type: "string" },
      chainId: { type: "string" },
      type: { type: "string" },
      nonce: { type: "string" },
      issuedAt: { type: "string", format: "date-time" },
      expirationTime: { type: "string", format: "date-time" },
      notBefore: { type: "string", format: "date-time" },
      requestId: { type: "string" },
      resources: { type: "array", items: { type: "string", format: "uri" } },
      signature: { type: "string" },
    },
    required: [
      "domain",
      "address",
      "uri",
      "version",
      "chainId",
      "type",
      "nonce",
      "issuedAt",
      "signature",
    ],
  };
}
